File: Day349/NumberOfLaserBeamsInABank.go
func numberOfBeams(bank []string) int {
    temp := 0
    answer := 0
    for _,v := range bank[0] {
        temp += int(v-'0')
    }
    for i:=1; i<len(bank); i++ {
        aTemp := 0
        for _,v := range bank[i] {
            aTemp += int(v-'0')
        }
        if aTemp != 0 {
            answer += aTemp*temp;
            temp = aTemp
        }
    }
    return answer
}

File: Day349/NumberOfLaserBeamsInABank.java
class Solution {
    public int numberOfBeams(String[] bank) {
        int temp = 0, answer = 0;
        for (char c: bank[0].toCharArray()) {
            temp += c-'0';
        }
        for (int i=1; i<bank.length; i++) {
            int pre = 0;
            for (char c : bank[i].toCharArray()) {
                pre += c-'0';
            }
            if (pre != 0) {
                answer += temp * pre;
                temp = pre;
            }
        }
        return answer;
    }
}